src/ex3/PokemonClient.js

```diff
@@ -0,0 +1,65 @@
+import fetch from 'node-fetch';
+import Image from 'ascii-art-image';
+
+export class PokemonClient {
+  constructor() {
+    this.API_URL = 'https://pokeapi.co/api/v2/pokemon/';
+  }
+
+  async fetchPokemons(ids) {
+    try {
+      const allResponses = await Promise.all(
+        ids.map((id) => fetch(`${this.API_URL}${id}/`))
+      );
+      const pokemons = await Promise.all(
+        allResponses.map((response) => response.json())
+      );
+
+      pokemons.forEach((pokemon) => {
+        this.printPokemonImage(
+          pokemon.sprites.other['official-artwork'].front_default
+        );
+      });
+      return pokemons;
+    } catch (err) {
+      console.log(err);
+      throw new Error('Failed to fetch pokemons');
+    }
+  }
+
+  async fetchSinglePokemon(id) {
```

Comment:
you can remove this method and use `fetchPokemons` when sending an array of one item, this way you will have less code duplication

Reply:
thanks, done!